packages/javascript/bh-shared-ui/src/components/SearchCurrentNodes/SearchCurrentNodes.tsx

```diff
@@ -0,0 +1,99 @@
+import { Box, List, ListItem, Paper, SxProps, TextField } from "@mui/material";
+import { useCombobox } from "downshift";
+import { FC, useEffect, useRef, useState } from "react";
+import SearchResultItem from "../SearchResultItem";
+import { FlatNode, GraphNodes } from "./types";
+import { useOnClickOutside } from "../../hooks";
+
+export const PLACEHOLDER_TEXT = "Search Current Results";
+export const NO_RESULTS_TEXT = "No result found in current results";
+
+const SearchCurrentNodes: FC<{
+    sx?: SxProps;
+    currentNodes: GraphNodes;
+    onSelect: (node: FlatNode) => void;
+    onClose?: () => void;
+}> = ({ sx, currentNodes, onSelect, onClose }) => {
+    
+    const containerRef = useRef<HTMLDivElement>(null);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const [items, setItems] = useState<FlatNode[]>([]);
+    const [selectedNode, setSelectedNode] = useState<FlatNode | null | undefined>(null)
+
+    // Node data is a lot easier to work with in the combobox if we transform to an array of flat objects
+    const flatNodeList: FlatNode[] = Object.entries(currentNodes).map(([key, value]) => {
+        return { id: key, ...value }
+    });
+
+    useEffect(() => inputRef.current?.focus(), []);
+
+    useEffect(() => {
+        if (selectedNode) onSelect(selectedNode);
+    }, [selectedNode, onSelect]);
+
+    useOnClickOutside(containerRef, () => onClose && onClose());
+
+    const { getInputProps, getMenuProps, getComboboxProps, getItemProps, inputValue } = useCombobox({
+        items,
+        onInputValueChange: ({ inputValue }) => {
+            const filteredNodes = flatNodeList.filter(node => {
+                const label = node.label.toLowerCase();
+                const objectId = node.objectId.toLowerCase();
+                const lowercaseInputValue = inputValue?.toLowerCase() || '';
+
+                if (inputValue === '') return false;
+
+                return label.includes(lowercaseInputValue) || objectId.includes(lowercaseInputValue);
+            });
+            setItems(filteredNodes);
```

Comment:
There is some, in my opinion, noticeable lag that can occur when changing this value. Do you reckon there is any easy fix we could consider?

Reply:
The changes you made make a night and day difference. Thanks for digging in!